python/lsst/daf/butler/core/dimensions/_coordinate.py

```diff
@@ -59,12 +60,20 @@
 DataCoordinate.
 """
 
-DataIdValue = Union[str, int, None]
+# Pydantic will cast int to str if str is first in the Union.
+DataIdValue = Union[int, str, None]
```

Comment:
Another duplicate of https://github.com/lsst/daf_butler/blob/f94a57823ca904996b74b8bab85282891f2d18a5/python/lsst/daf/butler/core/dimensions/_coordinate.py#L62

Reply:
It's the actual line but you have to put the int before the str to control Pydantic casting.

Reply:
Feel free to change the one in `_coordinate.py` to make pydantic happy; I'm pretty sure it's the only thing that cares about the order right now.

Reply:
We are commenting on the line in `_coordinate.py`. This change fixes the order.

Reply:
:facepalm: